lib/kernel.coffee

```diff
@@ -11,312 +11,31 @@ WatchSidebar = require './watch-sidebar'
-                data = data.toString()
-
-                console.log 'Kernel: stdout:', data
-
-                regexp = getKernelNotificationsRegExp()
-                if regexp?.test data
-                    atom.notifications.addInfo @kernelName,
-                        detail: data, dismissable: true
-
-            @kernelProcess.stderr.on 'data', (data) =>
-                data = data.toString()
-
-                console.log 'Kernel: stderr:', data
-
-                regexp = getKernelNotificationsRegExp()
-                if regexp?.test data
-                    atom.notifications.addError @kernelName,
-                        detail: data, dismissable: true
-
-    connect: ->
-        scheme = @config.signature_scheme.slice 'hmac-'.length
-        key = @config.key
-
-        @shellSocket = new jmp.Socket 'dealer', scheme, key
-        @controlSocket = new jmp.Socket 'dealer', scheme, key
-        @ioSocket    = new jmp.Socket 'sub', scheme, key
-
-        id = uuid.v4()
-        @shellSocket.identity = 'dealer' + id
-        @controlSocket.identity = 'control' + id
-        @ioSocket.identity = 'sub' + id
-
-        address = "#{ @config.transport }://#{ @config.ip }:"
-        @shellSocket.connect(address + @config.shell_port)
-        @controlSocket.connect(address + @config.control_port)
-        @ioSocket.connect(address + @config.iopub_port)
-        @ioSocket.subscribe('')
-
-        @shellSocket.on 'message', @onShellMessage.bind this
-        @ioSocket.on 'message', @onIOMessage.bind this
-
-        @shellSocket.on 'connect', -> console.log 'shellSocket connected'
-        @controlSocket.on 'connect', -> console.log 'controlSocket connected'
-        @ioSocket.on 'connect', -> console.log 'ioSocket connected'
-
-        try
-            @shellSocket.monitor()
-            @controlSocket.monitor()
-            @ioSocket.monitor()
-        catch err
-            console.error 'Kernel:', err
+    addWatchCallback: (watchCallback) ->
+        @watchCallbacks.push(watchCallback)
 
     interrupt: ->
-        console.log 'sending SIGINT'
-        unless @onlyConnect
-            @kernelProcess.kill('SIGINT')
-
-    # onResults is a callback that may be called multiple times
-    # as results come in from the kernel
-    _execute: (code, requestId, onResults) ->
-        header =
-                msg_id: requestId,
-                username: '',
-                session: '00000000-0000-0000-0000-000000000000',
-                msg_type: 'execute_request',
-                version: '5.0'
-
-        content =
-                code: code
-                silent: false
-                store_history: true
-                user_expressions: {}
-                allow_stdin: false
-
-        message =
-                header: header
-                content: content
-
-        @executionCallbacks[requestId] = onResults
-
-        @shellSocket.send new jmp.Message message
+        console.log 'Abstract method'
```

Comment:
`console.error 'Kernel: interrupt method not implemented'`

or

`throw new Error 'Kernel: interrupt method not implemented'`

I think throwing is better because we can always catch an exception.